packages/lib/server/getLuckyUser.ts

```diff
@@ -24,6 +24,14 @@ async function leastRecentlyBookedUser<T extends Pick<User, "id" | "email">>({
         },
         where: {
           eventTypeId,
+          attendees: {
+            some: {
+              noShow: false,
+            },
+          },
+          noShowHost: {
+            not: true,
```

Comment:
Can we have a comment in code here as to why we are using negation here?
Also, I think going forward we should update schema to add a default `false` for `noShowHost` like `attendee.noShow`

Reply:
As pointed out here https://github.com/prisma/prisma/issues/24252, not:true won't match `null` and thus it is filtering only the cases where `noShowHost` is explicitly `false`

CI doesn't have failing tests because we changed the default value of `noShow` in this PR itself. Because the tests are creating new bookings, it is `false` instead of null as it would be for all existing bookings in production. I discovered the bug accidentally because I didn't run the migrations and just ran the test.

Reply:
We should add tests for `noShowHost:null` scenario as well

Reply:
Yeah nice catch. Updated the where to explicitly check for null and false and added a test as well.